wallet/src/main/java/ch/admin/bag/covidcertificate/wallet/pdf/export/PdfExportFragment.kt

```diff
@@ -0,0 +1,121 @@
+			)
+		}
+	}
+
+	private val viewModel by viewModels<PdfViewModel>()
+
+	private var _binding: FragmentPdfExportBinding? = null
+	private val binding get() = _binding!!
+
+	private lateinit var certificateHolder: CertificateHolder
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+		certificateHolder = (arguments?.getSerializable(ARG_CERTIFICATE_HOLDER) as? CertificateHolder)
+			?: throw IllegalArgumentException("PDF export fragment created without a CertificateHolder!")
+	}
+
+	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+		_binding = FragmentPdfExportBinding.inflate(inflater, container, false)
+		return binding.root
+	}
+
+	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+		super.onViewCreated(view, savedInstanceState)
+		binding.toolbar.setNavigationOnClickListener { parentFragmentManager.popBackStack() }
+
+		binding.certificatePdfExportButton.setOnClickListener { viewModel.exportPdf(certificateHolder) }
+		binding.certificatePdfExportRetryButton.setOnClickListener { viewModel.exportPdf(certificateHolder) }
+
+		viewModel.pdfExportState.observe(viewLifecycleOwner) { onExportStateChanged(it) }
+	}
+
+	override fun onDestroyView() {
+		super.onDestroyView()
+		_binding = null
+	}
+
+	@SuppressLint("SetTextI18n")
+	private fun onExportStateChanged(state: PdfExportState) {
+		when (state) {
+			is PdfExportState.LOADING -> {
+				binding.certificatePdfExportLoadingIndicator.isVisible = true
+				binding.certificatePdfExportContent.isVisible = false
+			}
+			is PdfExportState.ERROR -> {
+				binding.certificatePdfExportLoadingIndicator.isVisible = false
+				binding.certificatePdfExportContent.isVisible = true
+				binding.certificatePdfExportIntroLayout.isVisible = false
+				binding.certificatePdfExportErrorLayout.isVisible = true
+				binding.certificatePdfExportTitle.setText(R.string.wallet_certificate_export_detail_error_title)
+				binding.certificatePdfExportErrorCode.text = state.error.code
+
+				if (state.error.code == ErrorCodes.GENERAL_OFFLINE) {
+					binding.certificatePdfExportStatusIcon.setImageResource(R.drawable.ic_no_connection)
+					val title = getString(R.string.wallet_certificate_export_detail_network_error_title)
+					val text = getString(R.string.wallet_certificate_export_detail_network_error_text)
+					binding.certificatePdfExportStatusText.text = "$title\n$text".makeSubStringBold(title)
+				} else {
+					binding.certificatePdfExportStatusIcon.setImageResource(R.drawable.ic_process_error)
+					val title = getString(R.string.wallet_certificate_export_detail_general_error_title)
+					val text = getString(R.string.wallet_certificate_export_detail_general_error_text)
+					binding.certificatePdfExportStatusText.text = "$title\n$text".makeSubStringBold(title)
+				}
+			}
+			is PdfExportState.SUCCESS -> {
+				val folder = File(requireContext().cacheDir, "certificate-pdfs")
```

Comment:
Personally, I'd define constant values for the folder-name, file-suffic and such